ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -958,9 +958,13 @@ protected ByteBuffer filterChunkToRange(ByteBuffer buf) {
         buf.position(0);
         buf.limit(0);
       } else {
-        long startOffsetInThisChunk = chunkIndex == 0 ? resolvedByteRange.getStartOffset() - offset : 0;
+        long relativeOffset = offset;
+        if (options.getBlobOptions.getBlobSegment() != -1) {
```

Comment:
use the constant, or alternatively add utility method GetBlobOptions.hasBlobSegment()

Reply:
I like the utility method idea, I'm going with that